cmd/cloud_sql_proxy/proxy.go

```diff
@@ -264,7 +264,9 @@ func parseInstanceConfig(dir, instance string, cl *http.Client) (instanceConfig,
 	if err != nil {
 		return instanceConfig{}, err
 	}
-	sql.BasePath = *host
+	if *host != "" {
+		sql.BasePath = *host
```

Comment:
I suggest using

```
sqladmin.NewClient(ctx, option.WithEndpoint(*host))
```

Then the user can actually set a host (e.g., "sqladmin.sandbox.googleapis.com") rather than a full URL path because of this CL:
https://code-review.googlesource.com/c/google-api-go-client/+/50190

(then you can remove the `!strings.HasSuffix(*host, "/")` check, and I'd also update the docs for the flag to include that it can be host[:port] or a full URL path)